alchemist/alchemist-implementationbase/src/main/kotlin/it/unibo/alchemist/model/interfaces/geometry/euclidean/twod/Segment2D.kt

```diff
@@ -14,84 +16,84 @@ data class Segment2D<P : Vector2D<P>>(val first: P, val second: P) {
     /**
      * Checks whether the segment is aligned to the x axis.
      */
-    val xAxisAligned: Boolean get() = MathUtils.fuzzyEquals(first.y, second.y)
+    val xAxisAligned: Boolean by lazy { MathUtils.fuzzyEquals(first.y, second.y) }
```

Comment:
This is actually way worse than computing every time. The cost is higher than a lazy delegate: is there a reason other than performance for the change?